### DISCUSSION TITLE
Caracal queries

### DISCUSSION THREAD

**Discussion by SanjanaSogimatt** [camicroscope/caMicroscope#785] [Q&A]
Hey, @birm I was running the caracal project locally. I somehow figured out what was wrong and what was outdated. But when I npm start the project, I get this error '`JsonWebTokenError: jwt must be provided`' and for some reason the `public key` is not being generated. And this runs only in git bash and not in powershell, Is it normal?  Would be great if anyone can help me with this

**Comment by SanjanaSogimatt** [comment]
And also @birm the resource targets which are mentioned in setup_script.sh file
   
```
    RESOURCE_IDX="https://raw.githubusercontent.com/camicroscope/Distro/master/config/mongo_idx.js"
    RESOURCE_COLLECTION="https://raw.githubusercontent.com/camicroscope/Distro/master/config/mongo_collections.js"
    RESOURCE_DEFAULT_DATA="https://raw.githubusercontent.com/camicroscope/Distro/master/config/default_data.js"
```
They give 404 error. What is supposed to be done in this case?

**Comment by SanjanaSogimatt** [comment]
@birm Here, I have used curl command instead of wget
  ```
    curl -sS "$RESOURCE_IDX" -o .seeder.idx.js
    curl -sS "$RESOURCE_DEFAULT_DATA" -o .seeder.default.js
    curl -sS "$RESOURCE_COLLECTION" -o .seeder.collection.js
```
Original was as below:
```
    wget -q RESOURCE_IDX -O .seeder.idx.js
    wget -q RESOURCE_DEFAULT_DATA -O .seeder.default.js
    wget -q RESOURCE_COLLECTION -O .seeder.collection.js
```

**Comment by birm** [comment]
If you run caracal with security on, you need a token (and thus a user).

Re the setup script, I haven't been keeping up with that. Honestly, I don't think we were expecting people to use it since most people come in from the distro which does those things automatically. In the past 3 years we moved some configuration/default data. I'll add an issue to rewrite the setup script!

Thanks for the catch!

**Reply by SanjanaSogimatt** [reply]
Got it!
